### PR TITLE
[#1034] Added Validation that entry rule is not a fragment

### PR DESCRIPTION
**Pull request by epoell**
Signed-off-by: Eva Poell <epoell@itemis.com>

### CODE REVIEW

File: org.eclipse.xtext.tests/src/org/eclipse/xtext/xtext/ValidationIssue1034Test.xtend [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+@InjectWith(XtextInjectorProvider)
+class ValidationIssue1034Test extends AbstractXtextTests {
+
+	@Inject extension ParseHelper<Grammar>
+	@Inject extension ValidationTestHelper
+
+	@Before
+	override setUp() throws Exception {
+		super.setUp();
+		with(XtextStandaloneSetup)
```
Comment (szarnekow) [inline org.eclipse.xtext.tests/src/org/eclipse/xtext/xtext/ValidationIssue1034Test.xtend]: Is the with-invocation necessary?

Comment (epoell) [reply org.eclipse.xtext.tests/src/org/eclipse/xtext/xtext/ValidationIssue1034Test.xtend]: It seems to me like it.
The tests fail without that line the tests fail.

File: org.eclipse.xtext/src/org/eclipse/xtext/xtext/XtextConfigurableIssueCodes.java [hunk 2]
```diff
@@ -67,6 +68,8 @@ protected void initialize(IAcceptor<PreferenceKey> acceptor) {
 		acceptor.accept(create(BIDIRECTIONAL_REFERENCE, SeverityConverter.SEVERITY_WARNING));
 		acceptor.accept(create(DISCOURAGED_RULE_NAME, SeverityConverter.SEVERITY_WARNING));
 		acceptor.accept(create(USAGE_OF_DEPRECATED_RULE, SeverityConverter.SEVERITY_WARNING));
+
```
Comment (kthoms) [inline org.eclipse.xtext/src/org/eclipse/xtext/xtext/XtextConfigurableIssueCodes.java]: nitpick: why empty line here?

File: org.eclipse.xtext/src/org/eclipse/xtext/xtext/XtextConfigurableIssueCodes.java [hunk 3]
```diff
@@ -47,6 +47,7 @@
 	public static final String INVALID_ANNOTAION = ISSUE_CODE_PREFIX + "InvalidAnnotation";
 	public static final String USAGE_OF_DEPRECATED_RULE = ISSUE_CODE_PREFIX + "UsageOfDeprecatedRule";
 	
+	public static final String INVALID_FRAGMENT_AS_FIRST_RULE = ISSUE_CODE_PREFIX + "InvalidFragmentFirstRule";
```
Comment (kthoms) [inline org.eclipse.xtext/src/org/eclipse/xtext/xtext/XtextConfigurableIssueCodes.java]: nitpick: why empty line before?

File: org.eclipse.xtext/src/org/eclipse/xtext/xtext/XtextValidator.java [hunk 4]
```diff
@@ -258,6 +258,13 @@ else if (GrammarUtil.isDatatypeRule((ParserRule) firstRule)) {
 					firstRule,
 					XtextPackage.Literals.ABSTRACT_RULE__NAME, 
 					ValidationMessageAcceptor.INSIGNIFICANT_INDEX);
+		} 
+		else if (((ParserRule) firstRule).isFragment()) {
```
Comment (kthoms) [inline org.eclipse.xtext/src/org/eclipse/xtext/xtext/XtextValidator.java]: nitpick: we usually place the `else` after the closing braces of the previous block

File: org.eclipse.xtext/src/org/eclipse/xtext/xtext/XtextConfigurableIssueCodes.java (hunk 2 above)
Comment (epoell) [reply org.eclipse.xtext/src/org/eclipse/xtext/xtext/XtextConfigurableIssueCodes.java]: I assumed the file was organised in semantic blocks and figured, that my new error code did not fit into the block above...
Would you prefer, to drop that empty line?

File: org.eclipse.xtext/src/org/eclipse/xtext/xtext/XtextConfigurableIssueCodes.java (hunk 3 above)
Comment (epoell) [reply org.eclipse.xtext/src/org/eclipse/xtext/xtext/XtextConfigurableIssueCodes.java]: Same as above.
Would you prefer to drop that empty line?

File: org.eclipse.xtext/src/org/eclipse/xtext/xtext/XtextValidator.java (hunk 4 above)
Comment (epoell) [reply org.eclipse.xtext/src/org/eclipse/xtext/xtext/XtextValidator.java]: Me too, I wanted to imitate the rest of the method.
Would you prefer it on the same line?

File: org.eclipse.xtext/src/org/eclipse/xtext/xtext/XtextConfigurableIssueCodes.java (hunk 3 above)
Comment (kthoms) [reply org.eclipse.xtext/src/org/eclipse/xtext/xtext/XtextConfigurableIssueCodes.java]: Sorry, I did not see that this was already the case above in the file. Imitating the file's existing style is right. But when I see this correctly, the blocks are "errors" and "warnings". So this should go the "errors" section.

File: org.eclipse.xtext/src/org/eclipse/xtext/xtext/XtextValidator.java (hunk 4 above)
Comment (kthoms) [reply org.eclipse.xtext/src/org/eclipse/xtext/xtext/XtextValidator.java]: This method is unfortunately an example where it is handled differently than in the rest of the file. So I'd say please change the one above also to match the file's common formatting.

File: org.eclipse.xtext/src/org/eclipse/xtext/xtext/XtextConfigurableIssueCodes.java (hunk 3 above)
Comment (epoell) [reply org.eclipse.xtext/src/org/eclipse/xtext/xtext/XtextConfigurableIssueCodes.java]: Thanks for the hint! I will change that

**Pull request merged**